pkg/stdlib/io/fs/write.go

```diff
@@ -0,0 +1,154 @@
+package fs
+
+import (
+	"context"
+	"os"
+	"sort"
+
+	"github.com/MontFerret/ferret/pkg/runtime/core"
+	"github.com/MontFerret/ferret/pkg/runtime/values"
+	"github.com/MontFerret/ferret/pkg/runtime/values/types"
+)
+
+// Write writes the given data into the file.
+// @params path (String) - path to file to write into.
+// @params data (Binary) - data to write.
+// @params params (Object) optional - additional parameters:
+//   * mode (String):
+//     * x - Exclusive: returns an error if the file exist. It can be
+//     combined with other modes
+//     * a - Append: will create a file if the specified file does not exist
+//     * w - Write (Default): will create a file if the specified file does not exist
+// @returns None
+func Write(_ context.Context, args ...core.Value) (core.Value, error) {
+	err := validateRequiredWriteArgs(args)
+	if err != nil {
+		return values.None, err
+	}
+
+	fpath := args[0].String()
+	data := args[1].(values.Binary)
+	params := defaultParams
+
+	if len(args) == 3 {
+		params, err = parseParams(args[2])
+		if err != nil {
+			return values.None, core.Error(
+				err,
+				"parse `params` argument",
+			)
+		}
+	}
+
+	flag, err := parseWriteMode(params.Mode)
```

Comment:
Why don't you parse it in ``parseParams`` function?
I think ``params`` struct should contain already parsed value.
What do you think about it?

Reply:
@ziflex good idea!